### PR TITLE
Update Top Sources Widget

### PR DESCRIPTION
**Pull request by kmshin1397**
[Clubhouse story #1236](https://app.clubhouse.io/astromarshal/story/1236/update-top-sources-widget)
Screenshot 2020-09-18 120042

- Match look of recent sources widget
- Switch from unbounded "sinceDaysAgo" within preferences widget to quick options for "24 hrs | 7 days | 30 days"

### CODE REVIEW

File: skyportal/handlers/api/classification.py [hunk 1]
```diff
@@ -163,6 +164,13 @@ def allowed_classes(hierarchy):
             self.current_user
         ):
             self.push_all(action='skyportal/FETCH_RECENT_SOURCES')
+
+        if classification.obj_id in map(
+            lambda view_obj_tuple: view_obj_tuple[1],
+            SourceViewsHandler.get_top_source_views_and_ids(self.current_user),
+        ):
+            self.push_all(action='skyportal/FETCH_TOP_SOURCES')
+        print(classification)
```
Comment (profjsb) [inline skyportal/handlers/api/classification.py:173]:
```suggestion
```

**Review by profjsb — CHANGES_REQUESTED** [review]
Looks great. Small edit.

File: static/js/components/TopSources.jsx [hunk 2]
```diff
@@ -1,64 +1,228 @@
...
+  fontWeight:
+    timespan.label === currentTimespan.label
+      ? theme.typography.fontWeightBold
+      : theme.typography.fontWeightMedium,
+});
+
+const timespans = [
+  { label: "PAST 24 HRS", sinceDaysAgo: "1" },
+  { label: "PAST 7 DAYS", sinceDaysAgo: "7" },
+  { label: "PAST 30 DAYS", sinceDaysAgo: "30" },
```
Comment (dannygoldstein) [inline static/js/components/TopSources.jsx:48]: Can we also add 6months and 1yr?

File: skyportal/tests/frontend/test_top_sources.py [hunk 3]
```diff
@@ -39,3 +41,50 @@ def test_top_sources(driver, user, public_source, public_group, upload_data_toke
...
+    assert data['data']['id'] == obj_id
+
+    twenty_days_ago = datetime.datetime.now() - datetime.timedelta(days=20)
+    sv = SourceView(
+        obj_id=obj_id,
+        username_or_token_id=upload_data_token,
+        is_token=True,
+        created_at=twenty_days_ago,
+    )
+    DBSession.add(sv)
```
Comment (dannygoldstein) [inline skyportal/tests/frontend/test_top_sources.py:75]: Why call the session factory directly instead of `DBSession()`?

**Review by dannygoldstein — APPROVED** [review]

File: static/js/components/TopSources.jsx (hunk 2 above)
Comment (kmshin1397) [reply static/js/components/TopSources.jsx]: Added. Also made option labels more concise and added tooltips for the detailed label:

Screenshot 2020-09-21 132354

File: skyportal/tests/frontend/test_top_sources.py (hunk 3 above)
Comment (kmshin1397) [reply skyportal/tests/frontend/test_top_sources.py]: Oops, yeah oversight on my part.

**Review by profjsb — APPROVED** [review]

**Pull request merged**